src/main/java/hudson/plugins/jira/selector/DefaultIssueSelector.java

```diff
@@ -27,64 +29,46 @@
 import hudson.scm.ChangeLogSet;
 import hudson.scm.ChangeLogSet.Entry;
 
-public final class DefaultIssueSelector extends AbstractIssueSelector {
+public class DefaultIssueSelector extends AbstractIssueSelector {
 
     private static final Logger LOGGER = Logger.getLogger(DefaultIssueSelector.class.getName());
 
     @DataBoundConstructor
     public DefaultIssueSelector() {
     }
 
+    /**
+     * See {@link #addIssuesRecursive(Run, JiraSite, TaskListener, Set)}
+     */
     @Override
     public Set<String> findIssueIds(@Nonnull final Run<?, ?> run, @Nonnull final JiraSite site,
             @Nonnull final TaskListener listener) {
-        return findIssueIdsRecursive(run, site.getIssuePattern(), listener);
+        HashSet<String> issuesIds = new HashSet<String>();
+        addIssuesRecursive(run, site, listener, issuesIds);
+        return issuesIds;
     }
 
     @Extension
     public static final class DescriptorImpl extends Descriptor<AbstractIssueSelector> {
 
         @Override
         public String getDisplayName() {
-            return Messages.DefaultUpdaterIssueSelector_DisplayName();
+            return Messages.DefaultIssueSelector_DisplayName();
         }
     }
 
+    protected Logger getLogger() {
+        return LOGGER;
+    }
+
     /**
      * Finds the strings that match JIRA issue ID patterns. This method returns
      * all likely candidates and doesn't check if such ID actually exists or
      * not. We don't want to use {@link JiraSite#existsIssue(String)} here so
      * that new projects in JIRA can be detected.
+     * 
      */
-    private static Set<String> findIssueIdsRecursive(Run<?, ?> build, Pattern pattern, TaskListener listener) {
-        Set<String> ids = new HashSet<String>();
-
-        // first, issues that were carried forward.
-        Run<?, ?> prev = build.getPreviousBuild();
-        if (prev != null) {
-            JiraCarryOverAction a = prev.getAction(JiraCarryOverAction.class);
-            if (a != null) {
-                ids.addAll(a.getIDs());
-            }
-        }
-
-        // then issues in this build
-        findIssues(build, ids, pattern, listener);
-
-        // check for issues fixed in dependencies
-        for (DependencyChange depc : RunScmChangeExtractor.getDependencyChanges(build.getPreviousBuild()).values()) {
-            for (AbstractBuild<?, ?> b : depc.getBuilds()) {
-                findIssues(b, ids, pattern, listener);
-            }
-        }
-        return ids;
-    }
-
-    /**
-     * @param pattern pattern to use to match issue ids
-     *            
-     */
-    protected static void findIssues(Run<?, ?> build, Set<String> ids, Pattern pattern, TaskListener listener) {
+    protected static void findIssues(Run<?, ?> build, Set<String> issueIds, Pattern pattern, TaskListener listener) {
```

Comment:
If i remove it, i will break binary compatibility... 
Also, tests are using it... Tests can be reworked to not use this. But binary compatibility might be a bigger issue.